src/utils/fetchClient.ts

```diff
@@ -0,0 +1,35 @@
+import { Person } from '../types';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const BASE_URL = 'https://mate-academy.github.io/react_people-table/api';
+
+function wait(delay: number) {
+  return new Promise(resolve => {
+    setTimeout(resolve, delay);
+  });
+}
+
+function request<T>(
+  url: string,
+  method = 'GET',
+): Promise<T> {
+  const options: RequestInit = { method };
+
+  return wait(100)
+    .then(() => fetch(BASE_URL + url, options))
+    .then(response => {
+      if (!response.ok) {
+        throw new Error();
+      }
+
+      return response.json();
+    });
+}
+
+export const client = {
+  get: <T>(url: string) => request<T>(url),
+};
+
+export const getPeople = () => {
+  return client.get<Person[]>('/people');
+};
```

Comment:
I can't understand, why you changed the file with the function given in the start code

with code below
```ts
import { Person } from './types/Person';

// eslint-disable-next-line max-len
const API_URL = 'https://mate-academy.github.io/react_people-table/api/people.json';

function wait(delay: number) {
  return new Promise(resolve => setTimeout(resolve, delay));
}

export function getPeople(): Promise<Person[]> {
  // keep this delay for testing purpose
  return wait(500)
    .then(() => fetch(API_URL))
    .then(response => response.json());
}
```